crates/biome_dependency_graph/src/module_visitor.rs

```diff
@@ -0,0 +1,385 @@
+                                   self.visit_identifier_binding(binding, jsdoc_comment.clone());
+                                }
+                            }
+                        },
+                        AnyJsObjectBindingPatternMember::JsObjectBindingPatternShorthandProperty(node) => {
+                            if let Ok(binding) = node.identifier() {
+                                if let Some(binding) = binding.as_js_identifier_binding() {
+                                   self.visit_identifier_binding(binding, jsdoc_comment.clone());
+                                }
+                            }
+                        },
+                        AnyJsObjectBindingPatternMember::JsBogusBinding(_) | AnyJsObjectBindingPatternMember::JsMetavariable(_) => {}
+                    }
+                }
+            }
+        }
+
+        Some(())
+    }
+
+    fn visit_identifier_binding(
+        &mut self,
+        binding: &JsIdentifierBinding,
+        jsdoc_comment: Option<String>,
+    ) -> Option<()> {
+        let name = binding.name_token().ok()?.token_text_trimmed();
+        self.register_export_with_name_and_jsdoc_comment(name, jsdoc_comment)
+    }
+
+    fn register_export_with_name_and_jsdoc_comment(
+        &mut self,
+        name: impl Into<Text>,
+        jsdoc_comment: Option<String>,
+    ) -> Option<()> {
+        self.data
+            .exports
+            .insert(name.into(), Export::Own(OwnExport { jsdoc_comment }));
+
+        Some(())
+    }
+
+    fn import_from_module_source(&self, module_source: AnyJsModuleSource) -> Option<Import> {
+        let specifier = module_source
+            .as_js_module_source()?
+            .inner_string_text()
+            .ok()?;
+        Some(self.import_from_specifier(specifier.text()))
+    }
+
+    fn import_from_specifier(&self, specifier: &str) -> Import {
+        let resolved_path =
+            self.resolver
+                .resolve(self.directory, specifier)
+                .and_then(|resolution| {
+                    Utf8PathBuf::from_path_buf(resolution.into_path_buf())
+                        .map_err(|path| ResolveError::NotFound(path.to_string_lossy().to_string()))
+                });
+        Import { resolved_path }
+    }
+}
+
+fn get_name(binding_result: AnyJsBinding) -> Option<TokenText> {
+    let name = binding_result
+        .as_js_identifier_binding()?
+        .name_token()
+        .ok()?
+        .token_text_trimmed();
+    Some(name)
+}
+
+fn get_ts_name(binding_result: AnyTsIdentifierBinding) -> Option<TokenText> {
+    let name = binding_result
+        .as_ts_identifier_binding()?
+        .name_token()
+        .ok()?
+        .token_text_trimmed();
+    Some(name)
+}
+
+/// Normalizes the text in a JSDoc comment by stripping the opening and trailing
+/// markers, and trimming the leading asterisk from every line.
+fn normalize_jsdoc_comment(text: &str) -> String {
```

Comment:
NIT: I would call this `trim_block_comment`. We don't know if it's a JSDoc because we don't check for any `@` property , it's just a normal comment at this point of the code 😅

Reply:
Comments that start with `/**` are **multiline comments** but they aren't called JSDoc. JSDoc are just special tags that have special meaning **inside a multiline comment**

Reply:
I posted the link below, but you might've missed it: https://jsdoc.app/about-getting-started

According to JSDoc themselves, this is a JSDoc-recognised documentation comment:

```
/** This is a description of the foo function. */
```

Inside a JSDoc comment, you can then have JSDoc tags, which are the ones like `@param` or `@return`. But JSDoc tags are ignored if they happen in an ordinary multiline comment such as those starting with `/*`.